dotnet/src/SemanticKernel/Orchestration/SKFunctionExtensions.cs

```diff
@@ -47,7 +36,8 @@ public static ISKFunction UseMaxTokens(this ISKFunction skFunction, int maxToken
     /// <returns>Self instance</returns>
     public static ISKFunction UseTemperature(this ISKFunction skFunction, double temperature)
     {
-        skFunction.RequestSettings.Temperature = temperature;
+        Verify.NotNull(skFunction.AIRequestSettings.CompleteRequestSettings, "Completion request settings cannot be empty");
```

Comment:
nit: "temperature cannot be null"

Might also want to have a check for "temperature must be between 0.0 and 1.0"